### PR TITLE
Mac developer setup using a VM

### PR DESCRIPTION
**Pull request by beausmith**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -135,3 +135,35 @@ Running the Python backend and React front-end with auto-reload:
...
+## Developer Setup on Mac with a Linux VM
+
+1.  Install Parallels.
+
+        brew cask install parallels
+
+2.  Download Ubuntu Desktop `.iso` image file: [ubuntu-18.04.4-desktop-amd64.iso](https://releases.ubuntu.com/18.04.4/ubuntu-18.04.4-desktop-amd64.iso)
+3.  Create Parallels VM using the Ubuntu iso image file.
+4.  Copy ssh public key `~/.ssh/id_rsa.pub` to VM `~/.ssh/authorized_keys`.
+5.  `git checkout` the project.
```
Comment (eventualbuddha) [inline README.md:148]:
```suggestion
5.  `git checkout` the project on the VM.
```

File: README.md [hunk 2]
```diff
@@ -117,21 +117,83 @@ This tool **does not** provide any features handled by other offline apps such a
...
+## Developer Setup on Mac with a Linux VM
+
+1.  Install Parallels.
+
+        brew cask install parallels
+
+2.  Download Ubuntu Desktop `.iso` image file: [ubuntu-18.04.4-desktop-amd64.iso](https://releases.ubuntu.com/18.04.4/ubuntu-18.04.4-desktop-amd64.iso)
+3.  Create Parallels VM using the Ubuntu iso image file.
+4.  Copy ssh public key `~/.ssh/id_rsa.pub` to VM `~/.ssh/authorized_keys`.
+5.  Install NVM (curl is an install dependency)
```
Comment (eventualbuddha) [inline README.md:142]: This should clearly indicate that this is to be run on the VM.

File: README.md [hunk 3]
```diff
@@ -117,21 +117,83 @@ This tool **does not** provide any features handled by other offline apps such a
...
+    ```
+    sudo apt install curl
+    curl -o- https://raw.githubusercontent.com/nvm-sh/nvm/v0.35.3/install.sh | bash
+    ```
+    Close terminal and reopen.
+    ```
+    nvm install 12
+    ```
+6.  Install `git`
```
Comment (eventualbuddha) [inline README.md:151]:
```suggestion
6.  Install `git`.
```

File: README.md [hunk 4]
```diff
@@ -117,21 +117,83 @@ This tool **does not** provide any features handled by other offline apps such a
...
+7.  `git clone` this project on the VM.
+8.  Follow steps above for "Running in Development"
```
Comment (eventualbuddha) [inline README.md:153]:
```suggestion
8.  Follow steps above for "Running in Development".
```

File: README.md [hunk 5]
```diff
@@ -117,21 +117,83 @@ This tool **does not** provide any features handled by other offline apps such a
...
+9.  Setup Github access to allow pushing to repos.
+
+### Optional Steps
+
+1. Install gpg to sign commits.
+2. Setup VS Code to use "Remote - SSH" extension to be able to edit files and access the terminal from within VS Code… and forward ports such that you can use a native browser as well.
```
Comment (eventualbuddha) [inline README.md:159]:
```suggestion
2. Setup VS Code to use ["Remote - SSH"](https://marketplace.visualstudio.com/items?itemName=ms-vscode-remote.remote-ssh) extension to be able to edit files and access the terminal from within VS Code… and forward ports such that you can use a native browser as well.
```

File: README.md [hunk 6]
```diff
@@ -117,21 +117,83 @@ This tool **does not** provide any features handled by other offline apps such a
...
+3. Setup dotfiles to your preference.
+
+### Setup SSH Host Shortcut
+
+Set up ssh host shortcut such that you can type `ssh vx` to connect to VM:
+
+1.  Open Settings in the VM (click the caret in the top right to reveal menu to click button to open "Settings")
```
Comment (eventualbuddha) [inline README.md:166]:
```suggestion
1.  Open Settings in the VM (click the caret in the top right to reveal menu to click button to open "Settings").
```

File: README.md [hunk 7]
```diff
@@ -117,21 +117,83 @@ This tool **does not** provide any features handled by other offline apps such a
...
+2.  Open Network in sidebar.
+3.  Click the gear next to the "Connected" in the "Wired" section.
+4.  Copy the IP address. Eg. `10.211.55.3`
```
Comment (eventualbuddha) [inline README.md:169]:
```suggestion
4.  Copy the IP address. Eg. `10.211.55.3`.
```

File: README.md [hunk 8]
```diff
@@ -117,21 +117,83 @@ This tool **does not** provide any features handled by other offline apps such a
...
+5.  Back on dev machine, create/open the file `~/.ssh/config` and add:
```
Comment (eventualbuddha) [inline README.md:170]: I think it's worth using the standard terminology here and using "host" and "guest" (or "VM").

File: README.md [hunk 9]
```diff
@@ -117,21 +117,83 @@ This tool **does not** provide any features handled by other offline apps such a
...
+### Troubleshooting
+
+#### Error message: `Error: ENOSPC: System limit for number of file watchers reached`
+
+[Increase the amount of watchers](https://github.com/guard/listen/wiki/Increasing-the-amount-of-inotify-watchers#the-technical-details):
+
+    echo fs.inotify.max_user_watches=524288 | sudo tee -a /etc/sysctl.conf
+    sudo sysctl -p
+
+### Cant find `pipenv`?
```
Comment (eventualbuddha) [inline README.md:186]:
```suggestion
### Can't find `pipenv`?
```

File: README.md [hunk 10]
```diff
@@ -117,21 +117,83 @@ This tool **does not** provide any features handled by other offline apps such a
...
+
+You may need to update your `PATH` to include `/.local/bin/`:
```
Comment (eventualbuddha) [inline README.md:188]:
```suggestion
You may need to update your `PATH` to include `~/.local/bin/`:
```

File: README.md [hunk 11]
```diff
@@ -117,21 +117,83 @@ This tool **does not** provide any features handled by other offline apps such a
...
+
+    export PATH="${HOME}/.local/bin:${PATH}"
+
+### Issues Installing Yarn via make?
```
Comment (eventualbuddha) [inline README.md:192]: Rather than adding this, can we fix the `Makefile`?

Comment (beausmith) [reply README.md:200]: Yes, but I'm gonna punt this to @benadida to do when the time comes.

File: README.md [hunk 12]
```diff
@@ -117,21 +117,91 @@ This tool **does not** provide any features handled by other offline apps such a
 
 On Linux for now.
```
Comment (eventualbuddha) [inline README.md:118]:
```suggestion
Requires Linux or Docker.
```

File: README.md [hunk 13]
```diff
@@ -117,21 +117,91 @@ This tool **does not** provide any features handled by other offline apps such a
...
+    ```
+2.  Install `git`.
+3.  `git clone` this project on the VM.
+4.  Follow steps above for "Running in Development".
+5.  Setup Github access to allow pushing to repos.
+
+### Optional Steps
+
+1. Install gpg to sign commits.
+2. Setup VS Code to use ["Remote - SSH"](https://marketplace.visualstudio.com/items?itemName=ms-vscode-remote.remote-ssh) extension to be able to edit files and access the terminal from within VS Code… and forward ports such that you can use a native browser as well.
```
Comment (eventualbuddha) [inline README.md:163]:
```suggestion
2. Setup VS Code to use ["Remote - SSH"](https://marketplace.visualstudio.com/items?itemName=ms-vscode-remote.remote-ssh) extension to be able to edit files and access the terminal from within VS Code… and forward ports such that you can use a browser on the host as well.
```

File: README.md [hunk 14]
```diff
@@ -117,21 +117,91 @@ This tool **does not** provide any features handled by other offline apps such a
...
+1.  Open Settings in the VM (click the caret in the top right to reveal menu to click button to open "Settings").
+2.  Open Network in sidebar.
+3.  Click the gear next to the "Connected" in the "Wired" section.
+4.  Copy the IP address. Eg. `10.211.55.3`.
+5.  On the host machine, create/open the file `~/.ssh/config` and add:
+    ```
+    Host vx
+      HostName 10.211.55.3
+    ```
+6.  Confirm this works by typing `ssh vx` to ssh into the VM. If your username is different on the VM than the host, you'll need to use `ssh USER@vx`.
```
Comment (eventualbuddha) [inline README.md:179]:
```suggestion
6.  Confirm this works by typing `ssh vx` to ssh into the VM. If your username is different on the VM than the host, add `User USER` to the ssh config for `vx` in  `~/.ssh/config`.
```

File: README.md [hunk 15]
```diff
@@ -117,21 +117,91 @@ This tool **does not** provide any features handled by other offline apps such a
...
+7.  Copy ssh public key fron the host machine `~/.ssh/id_rsa.pub` to the VM `~/.ssh/authorized_keys` for connecting without entering a password.
+    ```
+    `scp ~/.ssh/id_rsa.pub vx:~/.ssh/authorized_keys`
```
Comment (eventualbuddha) [inline README.md:182]:
```suggestion
cat ~/.ssh/id_rsa.pub | ssh vx 'cat >> ~/.ssh/authorized_keys'`
```

Comment (eventualbuddha) [reply README.md:182]: In case it's not clear why you'd want this, the `scp` version overwrites the file whereas the `ssh` one appends to it.

**Comment by kumar303** [comment]
Hmm, it looks like maybe renaming the `main` branch didn't catch this commit? I don't see it: https://github.com/votingworks/mail-ballot-manager/blob/main/README.md

**Comment by beausmith** [comment]
D'oh! Thanks @kumar303. I'll cherry-pick it.

**Comment by beausmith** [comment]
@kumar303 I found another missing commit too. 🙏

**Pull request merged**